cpp/arcticdb/pipeline/write_frame.cpp

```diff
@@ -157,7 +157,9 @@ write_frame(
     auto fut_slice_keys = slice_and_write(frame, slicing, get_partial_key_gen(frame, key), store, de_dup_map, sparsify_floats);
     // Write the keys of the slices into an index segment
     ARCTICDB_SUBSAMPLE_DEFAULT(WriteIndex)
-    return index::write_index(std::move(frame), std::move(fut_slice_keys), key, store);
+    return std::move(fut_slice_keys).thenValue([&frame, key, store](auto&& slice_keys){
```

Comment:
Capture `frame` explicitly by rvalue since it's getting moved, and others by reference

Reply:
I did the same for key, despite was not in one of your comments

Reply:
done